web/src/app/util/Time.tsx

```diff
@@ -0,0 +1,109 @@
+import { DateTime, Duration, DurationLikeObject } from 'luxon'
+import React, { useEffect, useState } from 'react'
+import {
+  formatTimestamp,
+  getDT,
+  getDur,
+  FormatTimestampArg,
+  formatRelative,
+} from './timeFormat'
+
+type TimeBaseProps = {
+  prefix?: string
+  suffix?: string
+}
+
+type TimeTimestampProps = TimeBaseProps &
+  Omit<FormatTimestampArg, 'time'> & {
+    time: string | null | undefined
+    zero?: string
+  }
+
+const TimeTimestamp: React.FC<TimeTimestampProps> = (props) => {
+  const [, setTS] = useState('') // force re-render
+  useEffect(() => {
+    if (!['relative', 'relative-date'].includes(props.format || '')) return
```

Comment:
Can we simplify this, or are strings with xyz_relative expected?

```suggestion
    if (props.format !== 'relative' || props.format !== 'relative-date' || !props.format) {
      return
    }
```

Reply:
It's testing if format === relative or relative-date (not substrings)